internal/factsengine/factsengine.go

```diff
@@ -21,17 +22,19 @@ type FactsEngine struct {
 }
 
 func NewFactsEngine(agentID, factsEngineService string) *FactsEngine {
+	fileSystem := afero.NewOsFs()
```

Comment:
So at the end we go with the DI hehe
If this is the case @rtorrero , could you take care of changing the rest of the gatherers that use the `CommandExecutor` so they use the same technique, and we pass the as variables.
I'm not a big fan of that, but hey, it looks like everyone else prefers this way.
You could do it of course in other PR